geoh5py/objects/curve.py

```diff
@@ -43,65 +43,53 @@ class Curve(CellObject):
         fields=(0x6A057FDC, 0xB355, 0x11E3, 0x95, 0xBE, 0xFD84A7FFCB88)
     )
 
-    def __init__(self, object_type: ObjectType, name="Curve", **kwargs):
+    def __init__(  # pylint: disable="too-many-arguments"
+        self,
+        object_type: ObjectType,
+        vertices: np.ndarray = ((0.0, 0.0, 0.0), (1.0, 1.0, 1.0)),
+        cells: np.ndarray | None = None,
+        current_line_id: uuid.UUID | None = None,
+        parts: np.ndarray | None = None,
+        name="Curve",
+        **kwargs,
+    ):
         self._current_line_id: uuid.UUID | None = None
-        self._parts: np.ndarray | None = None
-        super().__init__(object_type, name=name, **kwargs)
+        self._parts: np.ndarray | None
+        self._vertices: np.ndarray = self.validate_vertices(vertices)
+
+        if parts is not None:
+            if cells is not None:
+                raise ValueError("Parts can only be set if cells are not provided.")
```

Comment:
Trying to supply both at the same time. Either you make the cells from parts (on vertices), or cells directly. Ambiguous to know which one to use otherwise